api/src/main/java/org/openmrs/module/fgh/mpi/SnapshotEventProcessor.java

```diff
@@ -128,14 +138,70 @@ public void process(DatabaseEvent event) {
+					
+					if (isOpenCrMPI(mpiSystem)) {
+						Map<String, Object> fhirBundle = new HashMap(3);
+						fhirBundle.put(MpiConstants.FIELD_RESOURCE_TYPE, MpiConstants.BUNDLE);
+						fhirBundle.put(MpiConstants.FIELD_TYPE, MpiConstants.BATCH);
+						fhirBundle.put(MpiConstants.FIELD_ENTRY, fhirPatients);
+						
+						List<Object> response = mpiHttpClient.submitBundle("fhir", mapper.writeValueAsString(fhirBundle),
+						    List.class);
+						
+						successPatientCount = response.size() / 2;
+					} else if (isSanteMPI(mpiSystem)) {
+						Map<String, Object> fhirMessageHeaderEntry = generateMessageHeader();
+						
+						//The entry of resource in message bundle message
+						Map<String, Object> fhirResourceEntry = new HashMap<>(2);
+						
+						fhirResourceEntry.put("fullUrl", "metadata.epts.e-saude.net/bundle");
+						fhirResourceEntry.put("resource", new HashMap<>(3));
+						
+						getObjectOnMapAsMap("resource", fhirResourceEntry).put(MpiConstants.FIELD_RESOURCE_TYPE,
+						    MpiConstants.BUNDLE);
+						
+						getObjectOnMapAsMap("resource", fhirResourceEntry).put(MpiConstants.FIELD_TYPE,
+						    MpiConstants.FIELD_TYPE_HISTORY);
+						
+						List<Map<String, Object>> fhirPatintsPlusRequest = new ArrayList<Map<String, Object>>(
+						        fhirPatients.size());
+						
+						for (Map<String, Object> fhirPatient : fhirPatients) {
+							Map<String, Object> resourceEntry = getObjectOnMapAsMap(MpiConstants.FIELD_RESOURCE,
+							    fhirPatient);
+							
+							Object patientUuid = getObjectOnMapAsListOfMap(MpiConstants.FIELD_IDENTIFIER, resourceEntry)
+							        .get(0).get("value");
+							
+							Map<String, Object> patientEntry = fastCreateMap(MpiConstants.FIELD_RESOURCE, resourceEntry,
+							    "request", fastCreateMap("method", "POST", "url", "Parient/" + patientUuid), "fullUrl",
+							    "Parient/" + patientUuid);
+							
+							fhirPatintsPlusRequest.add(patientEntry);
+						}
+						
+						getObjectOnMapAsMap("resource", fhirResourceEntry).put(MpiConstants.FIELD_ENTRY,
+						    fhirPatintsPlusRequest);
+						
+						Map<String, Object> messageBundle = new HashMap<String, Object>();
+						messageBundle.put("resourceType", "Bundle");
+						messageBundle.put("type", "message");
+						messageBundle.put(MpiConstants.FIELD_ENTRY, new ArrayList<Map<String, Object>>(2));
+						
+						getObjectOnMapAsListOfMap(MpiConstants.FIELD_ENTRY, messageBundle).add(fhirMessageHeaderEntry);
+						getObjectOnMapAsListOfMap(MpiConstants.FIELD_ENTRY, messageBundle).add(fhirResourceEntry);
+						
+						Map<String, Object> response = mpiHttpClient.submitBundle("fhir/Bundle",
+						    mapper.writeValueAsString(messageBundle), Map.class);
+						
+						//Sante doen't return the amount of pacients successifuly received but if there is any error in one patient Sante throws Exception
+						//Force successPatientCount to be equal fhirPatients.size()
+						successPatientCount = fhirPatients.size();
```

Comment:
Instead of 'forcing' the count why don't you keep the logic specific to OpenCR if sante does not send back a response that tells you the success count

Reply:
Ok